### ISSUE TITLE
Bug: A property named "isA" yields a format error when camel case is selected

### ISSUE THREAD

**Issue by nrail-joerg-walter** [IBM/openapi-validator#167]
A property named "isA" seems to conform to camel case rules but it is flagged as not conforming.

**Comment by dpopp07** [comment]
Thanks for the issue. I think I agree that should be acceptable camel case. I'll need to change our regex that we use to determine conformance.

**Comment by mkistler** [comment]
This is a tricky one.  The general issue here is how to treat words that are a single character.  In the example given, the single character word comes at the end, but it wouldn't have to -- what should be the camelCase for "is_a_dog"?  You might say that it should be "isADog", but this has two adjacent capital letters, which also fails our standard camelCase regex's.

Given that there are only two valid single letter words in the English language ("a", and "I"), it might be possible to write a regex that allows for this, but disallows all other occurrences of adjacent capital letters.  And also allows either of these two to appear "bare" at the end of the string.

Here's an attempt:
```
> var exp = /^[a-z][a-z0-9]*([AI]{0,1}[A-Z][a-z0-9]+)*[AI]{0,1}$/
undefined
> exp.test("isA")
true
> exp.test("isADog")
true
> exp.test("isAADog")
false
> 
```

@nrail-joerg-walter Does this look like it would meet your needs?
